File: android/src/main/kotlin/br/com/elo7/flutter_elo_client/external/models/Params.kt
package br.com.elo7.flutter_elo_client.external.models

enum class Methods(val value: String) {
    GET("get"),
    PUT("put"),
    POST("post"),
    PATH("path"),
    DELETE("delete")
}

data class Params(
    val path: String,
    val body: Map<String, Any>? = null,
    val headers: Map<String, Any>? = null,
    val method: String = Methods.GET.value,
    val queryParams: Map<String, Any>? = null,
)

File: ios/Classes/external/models/Params.swift
enum Methods : String {
    case GET = "get"
    case PUT = "put"
    case POST = "post"
    case PATH = "path"
    case DELETE = "delete"
}

struct Params {
    var path: String
    var body: [String: Any]? = nil
    var headers: [String: Any]? = nil
    var method: String = Methods.GET
    var queryParams: [String: Any]? = nil

    init(
        path: String, 
        body: [String: Any]?, 
        headers: [String: Any]?,
        method: [String: Any]?,
        queryParams: [String: Any]?,
    ) {
        self.path = path
        self.body = body
        self.headers = headers
        self.method = method
        self.queryParams = queryParams
    }
}